keras_nlp/models/gpt2/gpt2_causal_lm_preprocessor.py

```diff
@@ -74,16 +74,17 @@ class GPT2CausalLMPreprocessor(GPT2Preprocessor):
     ds = ds.map(preprocessor, num_parallel_calls=tf.data.AUTOTUNE)
     """
 
-    def call(self, x, y=None, sample_weight=None):
+    def call(self, x, y=None, sample_weight=None, sequence_length=None):
         if y is not None or sample_weight is not None:
             logging.warning(
                 "`GPT2CausalLMPreprocessor` generates `y` and `sample_weight` "
                 "based on your input data, but your data already contains `y` "
                 "or `sample_weight`. Your `y` and `sample_weight` will be "
                 "ignored."
             )
-
-        x = super().call(x)
+        # Tokenize with one extra token to account for the truncation below.
+        sequence_length = (sequence_length or self.sequence_length) + 1
```

Comment:
This is a bit odd to me... `sequence_length` has a higher priority than `self.sequence_length` because we need to respect `max_length` in `generate` method, however for users not having this context, this could be weird. Can we override the `sequence_length` in `generate` method?

Reply:
like change the name from `max_length` to `sequence_length`? no strong preferences there.

let's avoid mutating config state on a layer, but not sure what the best approach is. this just seemed "least bad" to me